hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4133,9 +4156,61 @@ func reconcileVirtLauncherNetworkPolicy(policy *networkingv1.NetworkPolicy, hclu
 			},
 		},
 	}
-	policy.Spec.PodSelector = metav1.LabelSelector{
-		MatchLabels: map[string]string{
-			"kubevirt.io": "virt-launcher",
+	policy.Spec.Egress = []networkingv1.NetworkPolicyEgressRule{
+		{
+			To: []networkingv1.NetworkPolicyPeer{
+				{
+					// Allow access towards outside the cluster for the virtual machines (guest nodes),
+					// But deny access to other cluster's pods and services with the exceptions below
+					IPBlock: &networkingv1.IPBlock{
+						CIDR:   "0.0.0.0/0",
+						Except: blockedNetworks,
```

Comment:
Isn't this going to block the services clusterIP's ? Is something is exposed as a service I think guest cluster should be able to access

Reply:
the required services are listed in rules below, so access to them is allowed.

do you suggest we should grant the virtual machines access to all services exposed in the mgmt cluster?

Reply:
i think that if the destination is the virt-launcher pod (in the same HCP namespace), the traffic will be allowed by the pod selector term, because OVN "knows" that this service belongs to that pod.

this is same of the other pod selectors for apiserver/dns/ingress - i specify the podSelector, and traffic is allowed through the ServiceIP.